settings-portal/Settings.vala

```diff
@@ -0,0 +1,159 @@
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public
+ * License along with this library; if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor
+ * Boston, MA 02110-1335 USA.
+ */
+
+[DBus (name = "org.freedesktop.portal.Error")]
+public errordomain PortalError {
+    FAILED,
+    INVALID_ARGUMENT,
+    NOT_FOUND,
+    EXISTS,
+    NOT_ALLOWED,
+    CANCELLED,
+    WINDOW_DESTROYED
+}
+
+[DBus (name = "io.elementary.pantheon.AccountsService")]
+private interface Pantheon.AccountsService : Object {
+    public abstract int prefers_color_scheme { owned get; set; }
+}
+
+[DBus (name = "org.freedesktop.Accounts")]
+interface FDO.Accounts : Object {
+    public abstract string find_user_by_name (string username) throws GLib.Error;
+}
+
+private class AccountsServiceMonitor : GLib.Object {
+    private FDO.Accounts? accounts_service = null;
+    private Pantheon.AccountsService? pantheon_act = null;
+    private string user_path;
+
+    public int32 color_scheme { get; set; }
+
+    construct {
+        setup_user_path ();
+        setup_prefers_color_scheme ();
+    }
+
+    private void setup_user_path () {
+        try {
+            accounts_service = GLib.Bus.get_proxy_sync (
+                GLib.BusType.SYSTEM,
+               "org.freedesktop.Accounts",
+               "/org/freedesktop/Accounts"
+            );
+
+            user_path = accounts_service.find_user_by_name (GLib.Environment.get_user_name ());
+        } catch (Error e) {
+            critical (e.message);
+        }
+    }
+
+    private void setup_prefers_color_scheme () {
+        try {
+            pantheon_act = GLib.Bus.get_proxy_sync (
+                GLib.BusType.SYSTEM,
+                "org.freedesktop.Accounts",
+                user_path,
+                GLib.DBusProxyFlags.GET_INVALIDATED_PROPERTIES
+            );
+
+            color_scheme = pantheon_act.prefers_color_scheme;
+
+            ((GLib.DBusProxy) pantheon_act).g_properties_changed.connect ((changed, invalid) => {
+                var value = changed.lookup_value ("PrefersColorScheme", new VariantType ("i"));
+                if (value != null) {
+                    color_scheme = value.get_int32 ();
+                }
+            });
+        } catch (Error e) {
+            critical (e.message);
+        }
+    }
+}
+
+[DBus (name = "org.freedesktop.impl.portal.Settings")]
+public class SettingsDaemon.Settings : GLib.Object {
+    private GLib.DBusConnection connection;
```

Comment:
Hm, seems this was unnecessary after all.

Reply:
It's used for export the request on the others portals, no need for it here.

Reply:
Removed.